src/Compilers/CSharp/Test/Emit2/Semantics/PatternMatchingTests_ListPatterns.cs

```diff
@@ -8916,6 +8918,1143 @@ public void NotExhaustive_LongList()
+            {
+                string[] M(string[] s)
+                {
+                    if (s is ["(", ..{{typeSpecifier}} between, ")"])
+                    {
+                        return between;
+                    }
+
+                    return null;
+                }
+            }
+            """;
+
+        var comp = CreateCompilationWithIndexAndRangeAndSpan([source, TestSources.GetSubArray]);
+        comp.VerifyEmitDiagnostics();
+
+        CompileAndVerify(comp).VerifyIL("C.M", """
+            {
+              // Code size       73 (0x49)
+              .maxstack  4
+              .locals init (string[] V_0, //between
+                            int V_1)
+              IL_0000:  ldarg.1
+              IL_0001:  brfalse.s  IL_0047
+              IL_0003:  ldarg.1
+              IL_0004:  ldlen
+              IL_0005:  conv.i4
+              IL_0006:  stloc.1
+              IL_0007:  ldloc.1
+              IL_0008:  ldc.i4.2
+              IL_0009:  blt.s      IL_0047
+              IL_000b:  ldarg.1
+              IL_000c:  ldc.i4.0
+              IL_000d:  ldelem.ref
+              IL_000e:  ldstr      "("
+              IL_0013:  call       "bool string.op_Equality(string, string)"
+              IL_0018:  brfalse.s  IL_0047
+              IL_001a:  ldarg.1
+              IL_001b:  ldloc.1
+              IL_001c:  ldc.i4.1
+              IL_001d:  sub
+              IL_001e:  ldelem.ref
+              IL_001f:  ldstr      ")"
+              IL_0024:  call       "bool string.op_Equality(string, string)"
+              IL_0029:  brfalse.s  IL_0047
+              IL_002b:  ldarg.1
+              IL_002c:  ldc.i4.1
+              IL_002d:  ldc.i4.0
+              IL_002e:  newobj     "System.Index..ctor(int, bool)"
+              IL_0033:  ldc.i4.1
+              IL_0034:  ldc.i4.1
+              IL_0035:  newobj     "System.Index..ctor(int, bool)"
+              IL_003a:  newobj     "System.Range..ctor(System.Index, System.Index)"
+              IL_003f:  call       "string[] System.Runtime.CompilerServices.RuntimeHelpers.GetSubArray<string>(string[], System.Range)"
+              IL_0044:  stloc.0
+              IL_0045:  ldloc.0
+              IL_0046:  ret
+              IL_0047:  ldnull
+              IL_0048:  ret
+            }
+            """);
+    }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/60091")]
+    public void SlicePattern_DeclarationPatternOfNarrowedTypeInBetween_Array_CodeGen()
+    {
+        var source = """
+            class C
+            {
+                string[] M(object[] s)
+                {
+                    if (s is ["(", ..string[] between, ")"])
+                    {
+                        return between;
+                    }
+
+                    return null;
+                }
+            }
+            """;
+
+        var comp = CreateCompilationWithIndexAndRangeAndSpan([source, TestSources.GetSubArray]);
+        comp.VerifyEmitDiagnostics();
```

Comment:
This would be a good test to show side-effect. It's much harder to figure out from the IL block.

Reply:
Can you clarify what do you mean by "side-effect" here? We usually reffer to side effects as something, which is not a part of the main code flow or a potential unintentional behavior from user's perspective. The review is pointing to a test of an array, which by design should fully keep its original behavior with the only observable change being performance improvement in certain scenarios. So I wouldn't say there is a side effect of the optimization when dealing with arrays. Did you mean just to verify runtime behavior (which I'm gonna do anyway) or what?